tensorflow_lite_support/tools/ci_build/builds/build_ios_framework.sh

```diff
@@ -58,140 +60,51 @@ if [[ -z "${DEST_DIR+x}" || "${DEST_DIR}" == ${TFLS_ROOT_DIR}* ]]; then
   exit 1
 fi
 
-
-# Builds the C API framework for the specified framework for iOS.
-function build_c_api_framework {
-  "${BAZEL}" build -c opt --config=ios_fat \
-    //tensorflow_lite_support/ios:$1C_framework
+# get_output_file_path takes one bazel target label as an argument, and prints
+# the path of the first output file of the specified target.
+function get_output_file_path {
+  local STARLARK_FILE="${OUTPUT_TMPDIR}/print_output_file.starlark"
+  cat > "${STARLARK_FILE}" << EOF
+def format(target):
+  return target.files.to_list()[0].path
+EOF
+  bazel cquery --config=ios_fat $1 \
+    --output=starlark --starlark:file="${STARLARK_FILE}" 2> /dev/null
 }
 
-function create_framework_archive {
-  TARGET_FRAMEWORK_NAME="$1"
-  C_API_FRAMEWORK_NAME="$1C"
+function build_ios_api_framework {
+  local TARGET="//tensorflow_lite_support/ios"
+  FULL_TARGET="${TARGET}:${FRAMEWORK_NAME}_framework"
+
+  "${BAZEL}" build -c opt --config=ios_fat ${FULL_TARGET}
+  
+  OUTPUT_TMPDIR="$(mktemp -d)"
+  
+  # Find the path of the iOS framework generated by the above bazel build command
+  IOS_FRAMEWORK_PATH="$(get_output_file_path "${FULL_TARGET}")"
```

Comment:
This `IOS_FRAMEWORK_PATH` variable seems to be used in another function implicitly, which seems brittle.

Here's one suggestion to make this more robust:
* Make the `build_ios_api_framework` output the `IOS_FRAMEWORK_PATH` value (i.e., `echo ${IOS_FRAMEWORK_PATH}` at the end instead of printing out `OUTPUT_TMPDIR` which seems unnecessary)
* Make the `create_framework_archive` function to take the `IOS_FRAMEWORK_PATH` value as its first argument. (e.g., `IOS_FRAMEWORK_PATH="$1"`)
* After making these changes, you could call these functions at the end like this:
  ```sh
  ZIPPED_FRAMEWORK_PATH=$(build_ios_api_framework $FRAMEWORK_NAME)
  create_framework_archive "${ZIPPED_FRAMEWORK_PATH}"
  ```

Reply:
I have fixed this.